scapy/layers/bluetooth.py

```diff
@@ -1359,6 +1359,205 @@ def extract_padding(self, s):
+            0x5F: 'attachment:',
+            0x60: 'aw:',
+            0x61: 'barion:',
+            0x62: 'beshare:',
+            0x63: 'bitcoin:',
+            0x64: 'bolo:',
+            0x65: 'callto:',
+            0x66: 'chrome:',
+            0x67: 'chromeextension:',
+            0x68: 'comeventbriteattendee:',
+            0x69: 'content:',
+            0x6A: 'cvs:',
+            0x6B: 'dlnaplaysingle:',
+            0x6C: 'dlnaplaycontainer:',
+            0x6D: 'dtn:',
+            0x6E: 'dvb:',
+            0x6F: 'ed2k:',
+            0x70: 'facetime:',
+            0x71: 'feed:',
+            0x72: 'feedready:',
+            0x73: 'finger:',
+            0x74: 'fish:',
+            0x75: 'gg:',
+            0x76: 'git:',
+            0x77: 'gizmoproject:',
+            0x78: 'gtalk:',
+            0x79: 'ham:',
+            0x7A: 'hcp:',
+            0x7B: 'icon:',
+            0x7C: 'ipn:',
+            0x7D: 'irc:',
+            0x7E: 'irc6:',
+            0x7F: 'ircs:',
+            0x80: 'itms:',
+            0x81: 'jar:',
+            0x82: 'jms:',
+            0x83: 'keyparc:',
+            0x84: 'lastfm:',
+            0x85: 'ldaps:',
+            0x86: 'magnet:',
+            0x87: 'maps:',
+            0x88: 'market:',
+            0x89: 'message:',
+            0x8A: 'mms:',
+            0x8B: 'mshelp:',
+            0x8C: 'mssettingspower:',
+            0x8D: 'msnim:',
+            0x8E: 'mumble:',
+            0x8F: 'mvn:',
+            0x90: 'notes:',
+            0x91: 'oid:',
+            0x92: 'palm:',
+            0x93: 'paparazzi:',
+            0x94: 'pkcs11:',
+            0x95: 'platform:',
+            0x96: 'proxy:',
+            0x97: 'psyc:',
+            0x98: 'query:',
+            0x99: 'res:',
+            0x9A: 'resource:',
+            0x9B: 'rmi:',
+            0x9C: 'rsync:',
+            0x9D: 'rtmfp:',
+            0x9E: 'rtmp:',
+            0x9F: 'secondlife:',
+            0xA0: 'sftp:',
+            0xA1: 'sgn:',
+            0xA2: 'skype:',
+            0xA3: 'smb:',
+            0xA4: 'smtp:',
+            0xA5: 'soldat:',
+            0xA6: 'spotify:',
+            0xA7: 'ssh:',
+            0xA8: 'steam:',
+            0xA9: 'submit:',
+            0xAA: 'svn:',
+            0xAB: 'teamspeak:',
+            0xAC: 'teliaeid:',
+            0xAD: 'things:',
+            0xAE: 'udp:',
+            0xAF: 'unreal:',
+            0xB0: 'ut2004:',
+            0xB1: 'ventrilo:',
+            0xB2: 'viewsource:',
+            0xB3: 'webcal:',
+            0xB4: 'wtai:',
+            0xB5: 'wyciwyg:',
+            0xB6: 'xfire:',
+            0xB7: 'xri:',
+            0xB8: 'ymsgr:',
+            0xB9: 'example:',
+            0xBA: 'mssettingscloudstorage:'
+        }),
+        StrLenField('uri_hier_part', None, length_from=EIR_Element.length_from)
+    ]
+
+    @property
+    def uri(self):
+        return EIR_URI.scheme.i2s[self.scheme] + self.uri_hier_part.decode('utf-8')
```

Comment:
Did you try
```
return self.sprintf(%scheme%%uri_hier_part%")
```
I'm a bit unsure of whether this works on bytes but that's usually what we use.

Reply:
It seems to work for the byte part but not for the string haha

```
>>> p[EIR_URI].uri
"https:b'//firmware.silvair.com/oob'"
```